server/api/api/endpoints/nuclio.py

```diff
@@ -146,3 +162,438 @@ async def delete_api_gateway(
 
     async with server.api.utils.clients.async_nuclio.Client(auth_info) as client:
         return await client.delete_api_gateway(project_name=project, name=gateway)
+
+
+@router.post("/projects/{project}/functions/{name}/deploy")
```

Comment:
I think we might want to make it clear that this API is for nuclio only (to not confuse with building/deploying jobs). Why not have `nuclio` in the endpoint?
Maybe `/projects/{project}/nuclio/{name}/deploy`?